g_heap multiple definitions build error

Seeing this on many challenges when trying to compile on Arch.

```
[363/10141] Linking CXX executable challenges/CML/CML_patched
FAILED: challenges/CML/CML_patched 
: && /usr/bin/clang++  -m32 -z execstack -z norelro   -rdynamic challenges/CML/CMakeFiles/CML_patched.dir/lib/calloc.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/ctype.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/cxa.cc.o challenges/CML/CMakeFiles/CML_patched.dir/lib/exit.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/fflush.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/fopen.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/fread.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/free.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/fwrite.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/fxlat.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/malloc.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/malloc_common.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/memchr.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/memcmp.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/memcpy.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/memmove.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/memset.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/printf.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/realloc.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/stdio.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strcasecmp.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strcasestr.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strchr.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strcmp.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strcpy.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strdup.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strlen.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strncasecmp.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strncpy.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strndup.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strsep.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strstr.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strtol.c.o challenges/CML/CMakeFiles/CML_patched.dir/lib/strtoul.c.o challenges/CML/CMakeFiles/CML_patched.dir/src/attribute.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/cmlnode.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/interface.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/node.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/parser.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/query.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/service.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/string.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/stringmanager.cc.o challenges/CML/CMakeFiles/CML_patched.dir/src/textnode.cc.o -o challenges/CML/CML_patched  -Wl,-rpath,/home/rayce/builds/cb-multios/build/include:/home/rayce/builds/cb-multios/build/include/tiny-AES128-C  include/libcgc.so  include/tiny-AES128-C/libtiny-AES128-C.so && :
/usr/bin/ld: challenges/CML/CMakeFiles/CML_patched.dir/lib/free.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: multiple definition of `g_heap'; challenges/CML/CMakeFiles/CML_patched.dir/lib/calloc.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: first defined here
/usr/bin/ld: challenges/CML/CMakeFiles/CML_patched.dir/lib/malloc.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: multiple definition of `g_heap'; challenges/CML/CMakeFiles/CML_patched.dir/lib/calloc.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: first defined here
/usr/bin/ld: challenges/CML/CMakeFiles/CML_patched.dir/lib/malloc_common.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: multiple definition of `g_heap'; challenges/CML/CMakeFiles/CML_patched.dir/lib/calloc.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: first defined here
/usr/bin/ld: challenges/CML/CMakeFiles/CML_patched.dir/lib/realloc.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: multiple definition of `g_heap'; challenges/CML/CMakeFiles/CML_patched.dir/lib/calloc.c.o:/home/rayce/builds/cb-multios/build/../challenges/CML/lib/cgc_malloc_private.h:91: first defined here
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
[368/10141] Building C object challenges/CGC_Hangman_Game/CMakeFiles/CGC_Hangman_Game_patched.dir/src/words.c.o
ninja: build stopped: subcommand failed.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,6 +55,7 @@ else(WIN32)
     # Default flags for everything
     add_compile_options(
         -fno-builtin
+        -fcommon
         -w
         -g3
         -m32

